### DISCUSSION TITLE
Filter file not picking up my Match criteria

### DISCUSSION THREAD

**Discussion by gdeacon** [spotbugs/spotbugs-maven-plugin#472] [Q&A]
Hi Team,
I have been googling all over to figure out how to resolve this and have not found any suitable solution.  I'm trying to exclude some bug categories from the spotbugs analysis however the categories are still showing up in the report.  The filter file is very small here it is it is in the root project directory with the pom.xml (was not sure if this is the correct location)

<?xml version="1.0" encoding="UTF-8"?>
 <FindBugsFilter
              xmlns="https://github.com/spotbugs/filter/3.0.0"
              xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
              xsi:schemaLocation="https://github.com/spotbugs/filter/3.0.0 https://raw.githubusercontent.com/spotbugs/spotbugs/3.1.0/spotbugs/etc/findbugsfilter.xsd">
   <Match>
    <Bug category="I18N,BAD_PRACTICE" />
   </Match>

</FindBugsFilter>

I run this using the plugin using the following mvn command
mvn compile site -P codeQuality (codeQuality is my special profile for spotbug and PMD reports.

Here's the snippet of the plugin configuration...

  <profiles>
  	<profile>
		<id>codeQuality</id>
		  <reporting>
		    <plugins>
		      <plugin>
		        <groupId>org.apache.maven.plugins</groupId>
		        <artifactId>maven-pmd-plugin</artifactId>
		        <version>3.17.0</version>
				<configuration>
					<rulesets>
						<ruleset>/category/java/bestpractices.xml</ruleset>
						<ruleset>/category/java/codestyle.xml</ruleset>
						<ruleset>/category/java/design.xml/</ruleset>
						<ruleset>/category/java/errorprone.xml</ruleset>
						<ruleset>/category/java/multithreading.xml</ruleset>
						<ruleset>/category/java/performance.xml</ruleset>
						<ruleset>/category/java/security.xml</ruleset>
					</rulesets>	
					<format>net.sourceforge.pmd.renderers.SummaryHTMLRenderer</format>
					<renderSuppressedViolations>false</renderSuppressedViolations>
					<minimumPriority>3</minimumPriority>
				</configuration>
				<reportSets>
		          <reportSet>
		            <reports>
		              <report>pmd</report>
		            </reports>
		          </reportSet>
		        </reportSets>        
		      </plugin>
		      <plugin>
		        <groupId>org.apache.maven.plugins</groupId>
		        <artifactId>maven-project-info-reports-plugin</artifactId>
		        <version>3.3.0</version>
		        <reportSets>
		          <reportSet>
		            <reports>
		              <report>index</report>
		              <report>licenses</report>
		              <report>summary</report>
		              <report>modules</report>
		            </reports>
		          </reportSet>
		        </reportSets>
		      </plugin>
			  <plugin>
		         <groupId>org.apache.maven.plugins</groupId>
		         <artifactId>maven-jxr-plugin</artifactId>
		         <version>3.2.0</version>
		      </plugin>      
		      
			  <plugin>
                <groupId>com.github.spotbugs</groupId>
                <artifactId>spotbugs-maven-plugin</artifactId>
                <version>4.2.0</version>
				<configuration>
					<excludeFilterFile>spotbugs-exclude-filter.xml</excludeFilterFile>
 					<threshold>High</threshold>
    				<effort>Max</effort>
				</configuration>
              </plugin>
		    </plugins>
		  </reporting>

		<build>
			<plugins>
			  
			  <plugin>
			    <groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-site-plugin</artifactId>
				<version>4.0.0-M2</version>
			   </plugin>
		
			   <plugin>
				  <groupId>org.apache.maven.plugins</groupId>
				  <artifactId>maven-project-info-reports-plugin</artifactId>
				  <version>3.3.0</version>
				</plugin>  

				<plugin>
				        <groupId>org.apache.maven.plugins</groupId>
				        <artifactId>maven-pmd-plugin</artifactId>
				        <version>3.17.0</version>
						<configuration>
							<rulesets>
								<ruleset>/category/java/bestpractices.xml</ruleset>
								<ruleset>/category/java/codestyle.xml</ruleset>
								<ruleset>/category/java/design.xml/</ruleset>
								<ruleset>/category/java/errorprone.xml</ruleset>
								<ruleset>/category/java/multithreading.xml</ruleset>
								<ruleset>/category/java/performance.xml</ruleset>
								<ruleset>/category/java/security.xml</ruleset>
							</rulesets>	
							<format>net.sourceforge.pmd.renderers.SummaryHTMLRenderer</format>
							<renderSuppressedViolations>false</renderSuppressedViolations>
							<minimumPriority>3</minimumPriority>
						</configuration>

				</plugin>			
			    <plugin>
                   <groupId>com.github.spotbugs</groupId>
                   <artifactId>spotbugs-maven-plugin</artifactId>
                   <version>4.2.0</version>
					<configuration>
						<excludeFilterFile>spotbugs-exclude-filter.xml</excludeFilterFile>
	 					<threshold>High</threshold>
	    				<effort>Max</effort>
					</configuration>

                </plugin>
			
			</plugins>

		</build>

 	</profile>

**Comment by hazendaz** [comment]
being these are all reporting plugins, without some execution on them in your profile there on a normal build they won't do anything.  Look at the integration tests in this project as there are many examples.  Main thing is if you are running something like mvn clean install -PsomeProfile, you eend to also add execution to those plugins to get them to run.  If you are trying to  run mvn clean site -PsomeProfile as this case, you would also need ```<reporting>``` section.

**Reply by gdeacon** [reply]
Hi Jeremy,
Thank you for the quick response.  I'm not entirely sure what you mean about the fact that the plugin is in my reporting section of the profile.  I did see another error that I did not notice earlier.  Perhaps this will help.  It is failing to load the filter file.  It's showing the following stack trace...

I googled the error and some suggestions were to remove the line breaks on the schema setup in the FindBugsFilter element.  That did NOT fixe the error.  Any ideas what might cause the file not to be read?

Thaks

[INFO] Fork Value is true
     [java] The following errors occurred during analysis:
     [java]   Unable to read filter: /Users/lz3djn/Projects/storefront-aem/ally/ally-bundle-public/target/spotbugs-include-filter.xml : The processing instruction target matching "[xX][mM][lL]" is not allowed.
     [java]     java.io.IOException: The processing instruction target matching "[xX][mM][lL]" is not allowed.
     [java]       At edu.umd.cs.findbugs.filter.Filter.<init>(Filter.java:135)
     [java]       At edu.umd.cs.findbugs.FindBugs.configureFilter(FindBugs.java:471)
     [java]       At edu.umd.cs.findbugs.FindBugs2.addFilter(FindBugs2.java:404)
     [java]       At edu.umd.cs.findbugs.FindBugs2.configureFilters(FindBugs2.java:532)
     [java]       At edu.umd.cs.findbugs.FindBugs2.setUserPreferences(FindBugs2.java:505)
     [java]       At edu.umd.cs.findbugs.TextUICommandLine.configureEngine(TextUICommandLine.java:683)
     [java]       At edu.umd.cs.findbugs.FindBugs.processCommandLine(FindBugs.java:359)
     [java]       At edu.umd.cs.findbugs.FindBugs2.main(FindBugs2.java:1221)

**Comment by hazendaz** [comment]
maybe this setup here might help https://github.com/spotbugs/spotbugs-maven-plugin/tree/spotbugs/src/it/check-multi-filter-dups as it seems to be doing same thing and working.
